### PR TITLE
Preview TT functionality

### PR DESCRIPTION
**Pull request by AetherPrior**
Squashed all commits due to messy workflow into one commit:
* Add initial preview to timetable

* Make App stateful

* Add functionality for multiple hours displayed as a single block

* Add header rows/ columns, centre text

* Remove unnecessary code, fine tune

### CODE REVIEW

**Comment by AetherPrior** [comment]
The courses dictionary *can* be modified according to how @Aviral14 would find it convenient to send it over.

**Comment by AetherPrior** [comment]
Also, consider rebasing pull requests than merging?

File: client/src/App.js [hunk 1]
```diff
@@ -1,52 +1,98 @@
...
+import Timetable from "./components/TimeTable";
+
+
+class App extends Component {
+  constructor(props)
+  {
+    super(props);
+    this.state = {
+      isRoot: 1,
+      isPreview: 0
```
Comment (Kushagra-0801) [inline client/src/App.js:14]: there is a boolean type in JavaScript. Unless these were meant to be integers, consider changing them to bool

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    },
+        {
+            "type": "pwa-chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (Kushagra-0801) [inline .vscode/launch.json:23]: Not everyone might be using vscode, or have the same settings and all. Consider adding .vscode to `.gitignore` or `.git/info/exclude`

File: client/src/App.js [hunk 3]
```diff
@@ -1,52 +1,98 @@
...
+
+  }
+  sendPrevOrBack()
+  {
+    this.setState(state => ({
+      isPreview: !state.isPreview
+    }));
+    this.setState(state =>({
+      isRoot: !state.isRoot
+    }));
```
Comment (Kushagra-0801) [inline client/src/App.js:70]: `this.setState` works with a function too? This is new.
Anyways, these two calls, can be made into a single call, I think

File: client/src/components/TimeTable.jsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+         Map['Tuesday'] = 2;
+         //Map[2] = 'Tuesday';
+         Map['Wednesday'] = 3;
+         //Map[3] = 'Wednesday';
+         Map['Thursday'] = 4;
+         //Map[4] = 'Thursday';
+         Map['Friday'] = 5;
+         //Map[5] = 'Friday';
+         Map['Saturday'] = 6;
+         //Map[6] = 'Saturday';
```
Comment (Kushagra-0801) [inline client/src/components/TimeTable.jsx:99]: Remove useless commented code

File: client/src/components/TimeTable.jsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            
+            //let each course have a day list, and a hours list, both space separated strings
+            for(var course of coursesAdded)
+            {    
+                
+                var listOfDays = Array.from(course.Days.split(' '));
+                var listOfHours = (Array.from(course.Hours.split(' ')));
+                for(var i = 0; i< listOfHours.length; i++)
+                {
+                    listOfHours[i] = +listOfHours[i];
```
Comment (Kushagra-0801) [inline client/src/components/TimeTable.jsx:109]: this works?

File: client/src/components/TimeTable.jsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+         Map['Saturday'] = 6;
+         //Map[6] = 'Saturday';
+            
+            //let each course have a day list, and a hours list, both space separated strings
+            for(var course of coursesAdded)
+            {    
+                
+                var listOfDays = Array.from(course.Days.split(' '));
+                var listOfHours = (Array.from(course.Hours.split(' ')));
+                for(var i = 0; i< listOfHours.length; i++)
```
Comment (Kushagra-0801) [inline client/src/components/TimeTable.jsx:107]: use `let`, not `var`

File: client/src/components/TimeTable.jsx [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+                           gridRowEnd: `span ${listOfHours[listOfHours.length-1]-listOfHours[0]+1}`
+                        }
+                    }
+                    let str = <>
+                                <div className="gridItem" style={divStyle}>
+                                    {course.Name}<br></br>
+                                    {course.ID}<br></br>
+                                    {course.Section}<br></br>
+                                </div>
+                            </>;
```
Comment (Kushagra-0801) [inline client/src/components/TimeTable.jsx:129]: `<>`, `</>` are meant for when there isn't a single root iirc.
I don't think it is needed here

**Review by Kushagra-0801 — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (AetherPrior) [reply client/src/App.js:14]: Okay

File: .vscode/launch.json (hunk 2 above)
Comment (AetherPrior) [reply .vscode/launch.json:23]: Alright

File: client/src/App.js (hunk 3 above)
Comment (AetherPrior) [reply client/src/App.js:70]: I wouldn't do that if we need more than 2 different views. This is a temporary

File: client/src/components/TimeTable.jsx (hunk 4 above)
Comment (AetherPrior) [reply client/src/components/TimeTable.jsx:99]: We might need the commented code if the logic gets changed due to any reason.

File: client/src/components/TimeTable.jsx (hunk 5 above)
Comment (AetherPrior) [reply client/src/components/TimeTable.jsx:109]: Yes. And parseInt doesn't for some reason :/

File: client/src/components/TimeTable.jsx (hunk 7 above)
Comment (AetherPrior) [reply client/src/components/TimeTable.jsx:129]: Lite, but will change

File: client/src/components/TimeTable.jsx (hunk 4 above)
Comment (Kushagra-0801) [reply client/src/components/TimeTable.jsx:99]: That's what version control is for. Make a commit with commented code, and the final one without

**Comment by Kushagra-0801** [comment]
Just adding file in `.gitignore` doesn't remove it from the index.
you need to do `git rm --cache <file>` or something similar to remove file from index.

File: client/src/App.js [hunk 8]
```diff
@@ -1,52 +1,132 @@
...
+  {
+    super(props);
+    this.state = {
+<<<<<<< HEAD
+      isRoot: 1,
+      isPreview: 0
+=======
+      isRoot: true,
+      isPreview: false
+>>>>>>> feature
```
Comment (Kushagra-0801) [inline client/src/App.js:18]: You committed a merge conflict here

File: client/src/App.js [hunk 9]
```diff
@@ -1,52 +1,132 @@
...
+    }));
+   }
+   else if(view === this.view.PREVIEW)
+   {
+     this.setState(state => ({
+       isPreview: true,
+       isRoot: false
+     }))
+   }
+>>>>>>> feature
```
Comment (Kushagra-0801) [inline client/src/App.js:90]: Here too

**Comment by Kushagra-0801** [comment]
You committed a whole lot of merge conflicts. Check into them

**Comment by Kushagra-0801** [comment]
I suggest reverting the merge. Why did you even do it though?

File: client/src/components/TimeTable.jsx (hunk 5 above)
Comment (AetherPrior) [reply client/src/components/TimeTable.jsx:109]: Changed to include parseInt , have a look. I wasn't able to do it without a regular loop tho

File: client/src/App.js (hunk 3 above)
Comment (AetherPrior) [reply client/src/App.js:70]: Made another function. Just check it out

File: client/src/App.js (hunk 8 above)
Comment (AetherPrior) [reply client/src/App.js:18]: what... :o I'll fix it
It didn't throw any error while running

**Comment by AetherPrior** [comment]
I have done a git rm --cache as well. And I have absolutely no merge conflicts on my local copy. I have no idea what has happened, will check it in a bit

**Comment by AetherPrior** [comment]
> Why did you even do it though?  

are you okay with a pull request from the feature branch?

**Comment by AetherPrior** [comment]
There are too many issues here, am closing the pull request

**Pull request closed without merging**